lib/__tests__/standalone-cache.test.js

```diff
@@ -105,8 +109,8 @@ describe('standalone cache', () => {
 		expect(errored).toBe(true);
 
 		// Ensure only changed files are linted
-		expect(results.some((file) => file.source === validFile)).toBe(false);
-		expect(results.some((file) => file.source === newFileDest)).toBe(true);
+		expect(results.some((file) => isChanged(file, validFile))).toBe(false);
+		expect(results.some((file) => isChanged(file, newFileDest))).toBe(true);
 
 		const { cache } = fCache.createFromFile(expectedCacheFilePath);
 
```

Comment:
[suggest] I notice this changed test cases also pass on the `main` branch. So I guess the following case is necessary to test the fixed behavior. What do you think?

```js
describe('standalone cache uses a config file', () => {
	const cwd = path.join(__dirname, 'tmp', 'standalone-cache-use-config-file');

	safeChdir(cwd);

	const configFile = path.join(cwd, '.stylelintrc.json');
	const lintedFile = path.join(cwd, 'a.css');

	beforeEach(async () => {
		await fs.writeFile(lintedFile, 'a {}');
	});

	afterEach(async () => {
		await removeFile(configFile);
		await removeFile(lintedFile);
	});

	it('cache is discarded when a config file is changed', async () => {
		const config = { files: [lintedFile], config: undefined, cache: true };

		// No warnings when a rule is disabled.
		await fs.writeFile(configFile, JSON.stringify({ rules: { 'block-no-empty': null } }));
		const { results } = await standalone(config);
		expect(results[0].warnings.length).toBe(0);

		// Some warnings when a rule becomes enabled by changing the config.
		await fs.writeFile(configFile, JSON.stringify({ rules: { 'block-no-empty': true } }));
		const { results: resultsNew } = await standalone(config);
		expect(resultsNew[0].warnings.length).toBe(1);
	});
});
```

Reply:
Thanks. You are right. It was not enough to only refer to `ignored`.
I added a test you suggested.  https://github.com/stylelint/stylelint/pull/6356/commits/a3bff965057239859b87e9e9f400b43e975f22cd